src/test/java/ro/msg/learning/shop/strategy/impl/SingleLocationStrategyTest.java

```diff
@@ -0,0 +1,130 @@
+package ro.msg.learning.shop.strategy.impl;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import ro.msg.learning.shop.entity.*;
+import ro.msg.learning.shop.entity.compositeId.StockId;
+import ro.msg.learning.shop.repository.OrdersRepository;
+import ro.msg.learning.shop.service.*;
+import ro.msg.learning.shop.util.OrderDetailsUtil;
+import ro.msg.learning.shop.util.ProductQuantityPairUtil;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class SingleLocationStrategyTest {
+
+    @Mock
+    private OrdersRepository ordersRepository;
+    @Mock
+    private LocationService locationService;
+    @Mock
+    private ProductService productService;
+    @Mock
+    private StockService stockService;
+    @Mock
+    private OrderDetailService orderDetailService;
+    @Mock
+    private CustomerService customerService;
+
+    @InjectMocks
+    private SingleLocationStrategy singleLocationStrategy;
+
+    @Test
+    void handleOrderCreation() {
+        final Customer customer = new Customer();
```

Comment:
You can structure your code into more small methods in order to create the necessary mock data for the test. You can also create your mock data in a @BeforeEach method in order to reuse them in other tests. Have a look here: https://www.baeldung.com/junit-before-beforeclass-beforeeach-beforeall